### PR TITLE
Add env-level runs page

### PR DESCRIPTION
**Pull request by goodoldneon**
## Description
- Move function-level runs page into `FnRuns.tsx`. This is a copy-paste.
- Copy function-level runs page into `EnvRuns.tsx` and make the minimum changes to get it working for the env-level.
- Put new "Runs" nav item behind a feature flag.

There's a ton of code duplication between `FnRuns.tsx` and `EnvRuns.tsx`, but we'll DRY it out in a separate PR. Separating that into a later PR makes it easier to see the DRY changes

## Motivation
https://linear.app/inngest/issue/INN-3343

## Type of change (choose one)
- [ ] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/apps/dev-server-ui/src/app/(dashboard)/runs/page.tsx [hunk 1]
```diff
@@ -67,7 +70,7 @@ export default function Page() {
           durationMS,
         };
       });
-
+      console.log(edges.length);
```
Comment (djfarrelly) [inline ui/apps/dev-server-ui/src/app/(dashboard)/runs/page.tsx:73]:
```suggestion
```

File: ui/apps/dev-server-ui/src/app/(dashboard)/runs/page.tsx [hunk 2]
```diff
@@ -43,6 +43,9 @@ export default function Page() {
   const [startTime] = useSearchParam('start');
   const [endTime] = useSearchParam('end');
   const calculatedStartTime = useCalculatedStartTime({ lastDays, startTime });
+  console.log('calculatedStartTime', calculatedStartTime);
+  console.log('lastDays', lastDays);
+  console.log('startTime', startTime);
```
Comment (djfarrelly) [inline ui/apps/dev-server-ui/src/app/(dashboard)/runs/page.tsx:48]:
```suggestion
```

File: ui/apps/dashboard/src/components/Runs/EnvRuns.tsx [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+            endedAt
+            startedAt
+            status
+          }
+        }
+        pageInfo {
+          hasNextPage
+          hasPreviousPage
+          startCursor
+          endCursor
```
Comment (djfarrelly) [inline ui/apps/dashboard/src/components/Runs/EnvRuns.tsx:68]: **question (non-blocking)**: Would you recommend using fragments in the future?

**Review by djfarrelly — APPROVED** [review]
Approving so we can get this out and have users test it. It looks to be dependent on the new nav, so we'll have to check with Jacob on timing of rollout there./

@goodoldneon while you're in this code, it would be good to document that approach to DRYing out that view. My guess is that we need to create another component layer that handles all the requests given props like: GQL query documents and query values.

File: ui/apps/dashboard/src/components/Runs/EnvRuns.tsx (hunk 3 above)
Comment (goodoldneon) [reply ui/apps/dashboard/src/components/Runs/EnvRuns.tsx:68]: I think we should use a single query between both

File: ui/apps/dev-server-ui/src/app/(dashboard)/runs/page.tsx (hunk 2 above)
Comment (goodoldneon) [reply ui/apps/dev-server-ui/src/app/(dashboard)/runs/page.tsx:48]: I keep forgetting these 😭

**Review by jacobheric — APPROVED** [review]

**Pull request merged**